lib/editor-settings.php

```diff
@@ -81,3 +81,65 @@ function gutenberg_extend_post_editor_settings( $settings ) {
 	return $settings;
 }
 add_filter( 'block_editor_settings', 'gutenberg_extend_post_editor_settings' );
+
+/**
+ * Initialize a block-based editor.
+ *
+ * @param array $settings {
+ *      Elements to initialize a block-based editor.
+ *
+ *      @type array  $preload_paths        paths need to be preloaded.
+ *      @type string $editor_name          editor name.
+ *      @type string $editor_script_handle editor script handle.
+ *      @type string $initializer_name     editor initialization function name.
+ *      @type array  $editor_settings      editor settings.
+ * }
+ * @return void
+ */
+function gutenberg_initialize_editor( $settings ) {
+
+	$defaults = array(
+		'preload_paths'        => array(),
+		'editor_name'          => 'widgets_editor',
+		'editor_script_handle' => 'edit-widgets',
+		'initializer_name'     => 'initialize',
+		'editor_settings'      => array(),
+	);
+
+	$settings = wp_parse_args( $settings, $defaults );
+
+	$preload_paths = apply_filters( "{$settings['editor_name']}_preload_paths", $settings['preload_paths'] );
```

Comment:
Filter needs documentation.

For readability I'd also do:

```suggestion
	$editor_name = $settings['editor_name'];
	$preload_paths = apply_filters( "{$editor_name}_preload_paths", $settings['preload_paths'] );
```